src/main/java/com/iexec/common/chain/CredentialsAbstractService.java

```diff
@@ -50,6 +45,17 @@ public CredentialsAbstractService() throws Exception {
         }
     }
 
+    public CredentialsAbstractService(Credentials credentials) {
```

Comment:
Constructor in abstract classes should never be `public` as the constructor will never be called directly.
See Sonar alert and new code smell.

https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&sinceLeakPeriod=true&pullRequest=352&id=com.iexec.common%3Aiexec-common

Reply:
Updated here https://github.com/iExecBlockchainComputing/iexec-common/pull/352/commits/e57d5c61791cb1523c2b6bd010fe056ee5cc52d6, thanks for the spot!